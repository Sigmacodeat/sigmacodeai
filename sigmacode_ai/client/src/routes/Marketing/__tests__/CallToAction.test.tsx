import React from 'react';
import { render, screen } from '@testing-library/react';

// Test-lokaler i18n-Mock: respektiert defaultValue
jest.mock('react-i18next', () => {
  const actual = jest.requireActual('react-i18next');
  return {
    ...actual,
    useTranslation: () => ({
      t: (key: string, opts?: any) => (opts && opts.defaultValue) || key,
      i18n: { language: 'de', changeLanguage: jest.fn() },
    }),
  };
});

// IntersectionObserver Mock für framer-motion viewport/whileInView
class IO {
  observe() {}
  unobserve() {}
  disconnect() {}
}
// @ts-ignore
global.IntersectionObserver = IO as any;

import { Providers } from '../../../test/renderWithProviders';
import { CallToAction } from '../BusinessAI';

/**
 * Snapshot- und Rendering-Tests für die wiederverwendbare CallToAction-Komponente.
 * Prüft beide Varianten (hero, card) und die Motion-Option (nur Render, kein Animationsassert).
 */

describe('CallToAction', () => {
  const baseProps = {
    id: 'test-cta',
    title: 'Test Title',
    subtitle: 'Test Subtitle',
    primary: {
      to: '/primary',
      label: 'Primary',
      ariaLabel: 'Primary Label',
      titleAttr: 'Primary Title',
      analytics: 'primary-analytics',
    },
    secondary: {
      to: '/secondary',
      label: 'Secondary',
      ariaLabel: 'Secondary Label',
      titleAttr: 'Secondary Title',
      analytics: 'secondary-analytics',
    },
  } as const;

  it('renders hero variant with buttons', () => {
    const { container } = render(
      <Providers>
        <CallToAction {...baseProps} variant="hero" />
      </Providers>
    );

    expect(screen.getByRole('heading', { name: baseProps.title })).toBeInTheDocument();
    expect(screen.getByText(baseProps.subtitle)).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /primary/i })).toHaveAttribute('href', baseProps.primary.to);
    expect(screen.getByRole('link', { name: /secondary/i })).toHaveAttribute('href', baseProps.secondary!.to);
    expect(container).toMatchSnapshot();
  });

  it('renders card variant with motion enabled', () => {
    const { container } = render(
      <Providers>
        <CallToAction {...baseProps} variant="card" motion />
      </Providers>
    );

    // In card-Variante ist Title als sr-only vorhanden, Subtitle sichtbar
    expect(screen.getByText(baseProps.subtitle)).toBeInTheDocument();
    // Titel ist als sr-only vorhanden
    const srOnlyHeading = container.querySelector('h2.sr-only');
    expect(srOnlyHeading).not.toBeNull();
    // Buttons vorhanden
    expect(screen.getByRole('link', { name: /primary/i })).toHaveAttribute('href', baseProps.primary.to);
    expect(screen.getByRole('link', { name: /secondary/i })).toHaveAttribute('href', baseProps.secondary!.to);
  });
});
