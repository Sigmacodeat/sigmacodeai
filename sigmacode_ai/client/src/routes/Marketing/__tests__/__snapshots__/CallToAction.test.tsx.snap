// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction renders hero variant with buttons 1`] = `
<div>
  <section
    aria-labelledby="test-cta-title"
    class="py-20"
  >
    <div
      class="relative overflow-hidden rounded-2xl border border-gray-200 bg-gray-900 px-6 py-12 text-center text-gray-50 shadow-lg dark:border-gray-800 dark:bg-gray-900"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none absolute inset-0 -z-10"
      >
        <div
          class="absolute -top-20 left-1/2 h-64 w-64 -translate-x-1/2 rounded-full bg-gradient-to-tr from-teal-500/10 via-cyan-400/10 to-transparent blur-3xl"
        />
        <div
          class="absolute -bottom-24 left-16 h-56 w-56 rounded-full bg-gradient-to-tr from-indigo-500/10 via-fuchsia-500/10 to-transparent blur-3xl"
        />
        <div
          class="absolute -right-20 top-8 h-48 w-48 rounded-full bg-gradient-to-tr from-emerald-400/10 to-transparent blur-2xl"
        />
      </div>
      <h3
        class="text-2xl font-bold tracking-tight"
        id="test-cta-title"
      >
        Test Title
      </h3>
      <p
        class="mt-2 text-gray-300"
        id="test-cta-subtitle"
      >
        Test Subtitle
      </p>
      <div
        class="mt-6 flex flex-wrap justify-center gap-3"
      >
        <a
          aria-label="Primary Label"
          class="inline-flex items-center gap-2 rounded-lg px-4 py-2.5 text-sm font-medium text-white shadow-sm transition hover:shadow-md focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-0 dark:focus-visible:ring-offset-0 bg-gradient-to-r from-teal-600 to-cyan-600 hover:from-teal-500 hover:to-cyan-500 focus-visible:ring-teal-500/70 dark:focus-visible:ring-teal-300/60 px-3 py-2 text-xs sm:px-4 sm:py-2.5 sm:text-sm"
          data-analytics="primary-analytics"
          href="/primary"
          title="Primary Title"
        >
          Primary
        </a>
        <a
          aria-label="Secondary Label"
          class="inline-flex items-center gap-2 rounded-lg px-4 py-2.5 text-sm font-medium transition hover:shadow-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-0 dark:focus-visible:ring-offset-0 text-teal-300 border border-teal-400/30 bg-white/0 dark:bg-white/0 hover:bg-teal-500/10 focus-visible:ring-teal-500/70 dark:focus-visible:ring-teal-300/60 px-3 py-2 text-xs sm:px-4 sm:py-2.5 sm:text-sm"
          data-analytics="secondary-analytics"
          href="/secondary"
          title="Secondary Title"
        >
          Secondary
        </a>
      </div>
    </div>
  </section>
</div>
`;
